theano/gof/cmodule.py

```diff
@@ -2004,6 +2026,21 @@ def join_options(init_part):
         if GCC_compiler.march_flags:
             cxxflags.extend(GCC_compiler.march_flags)
 
+        # Find working march flag:
```

Comment:
We should first test the flags detected by Theano on line 2020. If they work, we use it. If not, try the alternative flags.